tools/tune/tdfb/bf_design.m

```diff
@@ -189,6 +197,18 @@
 end
 
 win = kaiser(bf.fir_length,bf.fir_beta);
+if isequal(DEBUG_SELT_FILT_ON,1)
+    win = hann (bf.fir_length);
+elseif isequal(DEBUG_SELT_FILT_ON,2)
+    win = hamming(bf.fir_length);
+elseif isequal(DEBUG_SELT_FILT_ON,3)
+    sidelobe = -30;
+    nbar = 4;
+    win = taylorwin(bf.fir_length, nbar, sidelobe);
+elseif isequal(DEBUG_SELT_FILT_ON,4)
+    sidelobe = 30;
+    win = chebwin(bf.fir_length, sidelobe);
+end
```

Comment:
The above could be a switch lower(bf_window), case 'kaiser', case 'hann', etc. The parameters for them could be like bf_kaiser_beta, bf_taylorwin_sidelobe, bf_taylorwin_nbar = 4, etc. You could set the bf_window to 'hann' in bf_defaults if that gave the best results.

Reply:
Sure !